Realm/RLMRealm.mm

```diff
@@ -944,16 +945,24 @@ - (RLMObject *)createObject:(NSString *)className withValue:(id)value {
 }
 
 - (BOOL)writeCopyToURL:(NSURL *)fileURL encryptionKey:(NSData *)key error:(NSError **)error {
+    return [self writeCopyToURL:fileURL encryptionKey:key enableHistory:NO error:error];
+}
+
+- (BOOL)writeCopyToURL:(NSURL *)fileURL encryptionKey:(NSData *)key enableHistory:(BOOL)enableHistory error:(NSError **)error {
     key = RLMRealmValidatedEncryptionKey(key);
     NSString *path = fileURL.path;
 
     try {
-        _realm->verify_thread();
-        try {
-            _realm->read_group().write(path.UTF8String, static_cast<const char *>(key.bytes));
-        }
-        catch (...) {
-            _impl::translate_file_exception(path.UTF8String);
+        if (enableHistory) {
+            _realm->write_copy(path.UTF8String, {static_cast<const char *>(key.bytes), key.length});
```

Comment:
If this is called on a different thread what happens?

Reply:
I believe it functions similarly. Ultimately the file is written in the format. And once that file is accessed, what thread it was written from is irrelevant. It's possible a developer could run into an error like calling writeCopy before all changes are synced to the server, or accessing a realm from the wrong thread.

I changed up `testSeedFilePathOpen` to write copy from a background thread:
```
            let queue = DispatchQueue(label: "background")
            queue.async {
                autoreleasepool {
                    do {
                        let realm = try Realm(configuration: config)
                        try realm.writeCopy(toFile: RLMTestRealmURL(), enableSync: true)
                    } catch {
                        XCTFail("Got an error: \(error)")
                    }
                }
            }
            queue.sync { }
```

I can keep it this way. I wrote another `testWriteCopyDifferentThread` but it was substantially similar testSeedFilePathOpen, and they take a bit of time to run. We could have `testSeedFilePathOpen ` just do the write copy from a background thread.

Reply:
The above will succeed because it was opened on the background thread in the first place. My concern is do we need to do similar with the non `enableSync` route and prepend `_realm->verify_thread()`.

Reply:
I'm not sure I understand. Since writeCopy is a method on the Realm, if the realm were being accessed from an incorrect thread, then the operation catches with "Realm accessed from incorrect thread".

Reply:
The verify thread check happens in Realm::write_copy.